### PR TITLE
contrib/net/http: add a convenience WrapClient method

### PR DESCRIPTION
**Pull request by gbbr**
This will facilitate integrating with libraries such as `go-github`

### CODE REVIEW

File: contrib/net/http/roundtripper.go [hunk 1]
```diff
@@ -66,3 +66,12 @@ func WrapRoundTripper(rt http.RoundTripper, opts ...RoundTripperOption) http.Rou
 		cfg:  cfg,
 	}
 }
+
+// WrapClient modifies the given client's transport to augment it with tracing and returns it.
+func WrapClient(c *http.Client, opts ...RoundTripperOption) *http.Client {
+	if c.Transport == nil {
+		c.Transport = http.DefaultTransport
```
Comment (bmermet) [inline contrib/net/http/roundtripper.go]: Is the `http.DefaultTransport` automatically wrapped?

Comment (gbbr) [reply contrib/net/http/roundtripper.go]: I'm not 100% sure I understand the question. If you're wondering if calling `WrapClient` will accidentally wrap the default transport, the answer is no.

**Comment by gbbr** [comment]
@dd-caleb good idea, check the latest commit - that's what you mean, right?

File: contrib/net/http/roundtripper.go (hunk 1 above)
Comment (bmermet) [reply contrib/net/http/roundtripper.go]: Indeed, I misread the code. My question doesn't make much sense.
Everything looks good 👍

**Pull request merged**